lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -337,17 +338,32 @@ def _connect_uncached(self):
             if self._play_context.private_key_file:
                 key_filename = os.path.expanduser(self._play_context.private_key_file)
 
-            ssh.connect(
-                self._play_context.remote_addr.lower(),
-                username=self._play_context.remote_user,
-                allow_agent=allow_agent,
-                look_for_keys=self.get_option('look_for_keys'),
-                key_filename=key_filename,
-                password=self._play_context.password,
-                timeout=self._play_context.timeout,
-                port=port,
-                **sock_kwarg
-            )
+            # paramiko 2.2 introduced auth_timeout parameter
+            if LooseVersion(paramiko.__version__) >= LooseVersion('2.2.0'):
+                ssh.connect(
+                    self._play_context.remote_addr.lower(),
+                    username=self._play_context.remote_user,
+                    allow_agent=allow_agent,
+                    look_for_keys=self.get_option('look_for_keys'),
+                    key_filename=key_filename,
+                    password=self._play_context.password,
+                    timeout=self._play_context.timeout,
+                    auth_timeout=self._play_context.timeout,
+                    port=port,
+                    **sock_kwarg
+                )
+            else:
```

Comment:
I think instead of having `ssh.connect` in an if/else, you should instead rename `sock_kwarg` to something like `kwargs` and conditionally add `auth_timeout` to `kwargs`

Reply:
Agree. Will do.

Reply:
@sivel done.